docs/content/guides/operator/sui-full-node.mdx

```diff
@@ -79,6 +79,41 @@ clang \
 cmake
 ```
 
+### Considerations to enable gRPC (beta)
+
+To serve the [gRPC API in beta](/concepts/grpc-overview.mdx), you must enable it on your Full node by indexing gRPC-specific data. During the initial gRPC indexing phase, your Full node may not be able to handle other traffic — including the [JSON-RPC](/references/sui-api.mdx) requests. Plan your rollout carefully (such as using the `rolling upgrade` mechanism), and communicate any downtime to your customers and partners in advance. The time required to sync gRPC indexes depends on your Full node's hardware and software configuration.
+
+Enable gRPC indexing on your Full node using the following configuration:
+
+```sh
+
+rpc:
+  enable-indexing: true
+
+```
+
+Running gRPC alongside JSON-RPC will increase your Full node’s storage usage. While this is expected in the short term, you can reclaim this storage in the future by disabling JSON-RPC after it is phased out. Developers must gradually migrate to gRPC or [GraphQL RPC](/concepts/graphql-rpc.mdx) (which will internally rely on gRPC). You can also run gRPC and JSON-RPC on separate nodes to optimize resource usage and provide a performant plus resilient RPC experience to applications. Use the following config to selectively disable JSON-RPC on a Full node:
```

Comment:
```suggestion
Running gRPC alongside JSON-RPC increases your Full node’s storage usage. While this is expected in the short term, you can reclaim this storage in the future by disabling JSON-RPC after it is phased out. You must gradually migrate to gRPC or [GraphQL RPC](/concepts/graphql-rpc.mdx) (which internally relies on gRPC). You can also run gRPC and JSON-RPC on separate nodes to optimize resource usage and provide a performant and resilient RPC experience to applications. Add the following to your fullnode.yaml configuration to selectively disable JSON-RPC on a Full node:
```

Reply:
@ronny-mysten - I'm taking bits of this. Basically the migration comment has to be focused on "Developers" as this doc is for full node operator config and "You" may not be compatible. Also, GraphQL doesn't use gRPC yet but will in future, hence "will rely" for now is appropriate.